### ISSUE TITLE
[Faction Branch] Unnecessary short hallway between husk modules and other modules

### ISSUE THREAD

**Issue by Caithloki** [Regalis11/Barotrauma#11206]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

The Husk quarter doubles up on doors, so you need to open two since both prefabs are butted up to each other directly.
![image](https://user-images.githubusercontent.com/12504812/218223654-3498e0ec-0c7c-4a9d-8a44-92eb25a868ee.png)

### Reproduction steps

Enter outpost
find husk quarter that is top/down connecting
there will possibly be two doors

### Bug prevalence

Happens regularly

### Version

Faction/endgame test branch

### Which operating system did you encounter this bug on?

Windows

**Comment by ubertpendragon** [comment]
Note that in outpost/ruin generation the lack of minimum Hallway length between modules always feel preferable because it keeps outposts/ruins compact and much less "clicky" (there are a LOT of doors in ruins)

The only thing holding this back is that regular gaps cannot be linked to waypoints unlike doors; if the side of a module isn't connected, it deletes the door and the waypoint associated to it. This isnt the case with gaps and it messes with the pathfinding, connecting waypoints outside.

Pictured: doorless gap not connected 
![image](https://user-images.githubusercontent.com/73229309/218226241-e534b14e-a25f-49c9-a988-8a98ecad707a.png)


In short to get rid of double doors at no minimum hallway length you'd need gaps deleting waypoints if not connected

**Comment by Regalis11** [comment]
Fixed in https://github.com/Regalis11/Barotrauma-development/commit/6155f978574c1f5ddd9c5e57560c4be8b172babc

**Comment by ubertpendragon** [comment]
> Outpost generator only takes walls with a collider into account when determining the bounds of the modules. Fixes husk modules being placed unnecessarily far from the outpost due to the decorative structures outside the door, leaving a very short hallway between the modules. Regalis11/Barotrauma#11206

Don't forget about gaps not deleting waypoints if not connected to another module (which breaks NPC AI following waypoints that go inside a wall and outside like on the ruins screenshot)

**Comment by ubertpendragon** [comment]
🙏  hoping unconnected gaps (not just doors) will also delete waypoints, that would allow for doorless modules and much less doors in outposts/ruins in general without breaking AI

**Comment by Rokvach** [comment]
Tested, working correctly. Closing.
